osgp/platform/osgp-secret-management/src/main/java/org/opensmartgridplatform/secretmgmt/application/services/encryption/DefaultEncryptionDelegate.java

```diff
@@ -0,0 +1,50 @@
+package org.opensmartgridplatform.secretmgmt.application.services.encryption;
+
+import org.opensmartgridplatform.secretmgmt.application.services.encryption.providers.JreEncryptionProvider;
+import org.opensmartgridplatform.secretmgmt.application.services.encryption.providers.EncryptionProvider;
+import org.opensmartgridplatform.secretmgmt.application.services.encryption.providers.EncryptionProviderType;
+import org.opensmartgridplatform.secretmgmt.application.services.encryption.providers.HsmEncryptionProvider;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * This class is used to encrypt/decrypt typed secrets by either the HardwareSecurityModule or by the JRE
+ */
+public class DefaultEncryptionDelegate implements EncryptionDelegate {
+
+    private final List<EncryptionProvider> providers = new ArrayList<>();
+
+    public DefaultEncryptionDelegate(File jreKeyFile) throws Exception {
+        this(jreKeyFile, null);
+    }
+
+    public DefaultEncryptionDelegate(File jreKeyFile, File hsmKeyStoreFile) throws Exception {
+        JreEncryptionProvider jreEncryptionProvider = new JreEncryptionProvider();
+        jreEncryptionProvider.setKeyFile(jreKeyFile);
+
+        providers.add(jreEncryptionProvider);
```

Comment:
this seems a strange construct and not open to extension. Why not create a second Provider for HSM?